src/filtered-list/index.ts

```diff
@@ -0,0 +1,67 @@
+import { tag, Tag } from '@storefront/core';
+import { DEFAULT_INDEX_ALIAS, DEFAULT_ITEM_ALIAS } from '../list';
+
+@tag('gb-filtered-list', require('./index.html'))
+class FilteredList {
+
+  refs: { filter: HTMLInputElement };
+  items: FilteredList.Item[];
+  originalItems: FilteredList.Item[];
+  props: FilteredList.Props = {
+    items: [],
+    itemAlias: DEFAULT_ITEM_ALIAS,
+    indexAlias: DEFAULT_INDEX_ALIAS,
+  };
+
+  onMount() {
+    this.originalItems = this.props.items;
+    this.updateItems();
+  }
+
+  onUpdate() {
+    if (this.props.items !== this.originalItems) {
+      this.originalItems = this.props.items;
+      this.updateItems();
+    }
+  }
+
+  onFilterChange(event: Tag.Event) {
+    event.preventUpdate = true;
+    this.updateItems();
+  }
+
+  updateItems() {
+    const value = this.refs.filter.value.trim().toLowerCase();
+    const items = this.props.items.filter((item) => {
+      if (!item) {
+        return false;
+      } else if (typeof item === 'string') {
+        return item.toLowerCase().includes(value);
+      } else if (typeof item.value === 'string') {
+        return item.value.toLowerCase().includes(value);
+      } else {
+        return false;
+      }
+    });
+    this.update({ items });
```

Comment:
instead of updating here (and having the potential of an update loop) we can just have it set `this.items` at the end, my bad